### PR TITLE
[#973] improvement: Make shuffle manager client RPC timeout configurable

### PR DESCRIPTION
**Pull request by cchung100m**
### What changes were proposed in this pull request?

Make shuffle manager client RPC timeout configurable

### Why are the changes needed?

Fix: #973 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

current UT and Integration testing

### CODE REVIEW

File: common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java [hunk 1]
```diff
@@ -129,6 +129,12 @@ public class RssBaseConf extends RssConf {
           .defaultValue(ClientType.GRPC)
           .withDescription("client type for rss");
 
+  public static final ConfigOption<Long> RSS_CLIENT_TYPE_GRPC_TIMEOUT =
+      ConfigOptions.key("rss.rpc.client.type.grpc.timeout")
+          .longType()
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java:134]: Could we use the method checkValue to validate the config option?

File: common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java [hunk 2]
```diff
@@ -129,6 +129,12 @@ public class RssBaseConf extends RssConf {
           .defaultValue(ClientType.GRPC)
           .withDescription("client type for rss");
 
+  public static final ConfigOption<Long> RSS_CLIENT_TYPE_GRPC_TIMEOUT =
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java:132]: `RSS_CLIENT_TYPE_GRPC_TIMEOUT_MS` may be a better name.

File: common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java (hunk 1 above)
Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java:134]: +1

Comment (cchung100m) [reply common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java:134]: Hi @jerqi  @zuston 

Thanks for the prompt reply and I updated the part you mentioned.

File: common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java [hunk 3]
```diff
@@ -129,6 +129,13 @@ public class RssBaseConf extends RssConf {
           .defaultValue(ClientType.GRPC)
           .withDescription("client type for rss");
 
+  public static final ConfigOption<Long> RSS_CLIENT_TYPE_GRPC_TIMEOUT_MS =
+      ConfigOptions.key("rss.rpc.client.type.grpc.timeout")
+          .longType()
+          .checkValue(ConfigUtils.POSITIVE_INTEGER_VALIDATOR, "The value must be positive integer")
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java:135]: Could we use `POSITIVE_LONG_VALIDATOR` here?

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zuston @cchung100m

**Review by zuston — APPROVED** [review]
LGTM

**Comment by rickyma** [comment]
I don't think this works as expected, because we always new an `RssBaseConf`.

**Pull request merged**